hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -64,43 +97,50 @@
     { name: "gpio",
       type: "gpio",
       clock_connections: {clk_i: "fixed"},
+      clock_group: "peri",
```

Comment:
having `clock_connections` and `clock_group` both looks a little bit awkward to me. But also it may reduce the number of letters if a IP has multiple clocks? for example, 

`clk_i <- clk_fixed_peri , clk_usb_48mhz_i <- clk_usb_peri `

Am I understanding correctly?

Reply:
yes.

Basically the question is whether we want to type ALL of this out, or we are comfortable letting topggen generate them.

What's happening now is effectively `top_earlgrey` is specifying what group each peripheral belongs to, and which source should be connected to each port.  Topgen then makes the modifications and connections.

Reply:
so imagine for the crypt block. They are supposed to be transactional and individually gate-able. So this means we could either manually type out `clk_main_aes`, `clk_main_hmac`, `clk_main_kmac`, `clk_main_bignum`, `clk_main_keymgr`, or we specify that each block has a source of `main`, belongs to clock group `trans`, and then each clock is individually created.

Honestly both are....okay to me, that's why i wanted some extra input.

Reply:
I think the mechanism to automatically generate all these gated clocks is the right approach (I'd rather automate this than hardcode). But the field name `clock_connections` may be ill named in this case. Wouldn't something like `clock_root` be more concise? topgen could then still derive the clocks and create the `clock_connections` in the output...